Core/src/Upload/ResumableUploader.php

```diff
@@ -139,8 +141,11 @@ public function resume($resumeUri)
     /**
      * Triggers the upload process.
      *
+     * Errors are of form [`google.rpc.Status`](https://cloud.google.com/apis/design/errors#error_model),
+     * and may be obtained via {@see Google\Cloud\Core\Exception\ServiceException::getMetadata()}.
+     *
      * @return array
-     * @throws GoogleException
+     * @throws ServiceException
```

Comment:
Was this a typo in documentation or was it recently changed?

At the moment, I don't suspect it to be an issue.

Reply:
It was changed because ServiceException includes utilities for exposing the underlying error to users, while GoogleException does not. Since the uploader does not provide the underlying error detail, and we cannot change the message without potentially causing issues for users who may be relying on it, we swapped the type out. ServiceException extends GoogleException, so we can replace it without breaking changes.

Reply:
Thanks for clarifying, this makes sense.